From e7ce2130089a690babb3f0082d55c9d8b37be9f8 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Mon, 06 Apr 2020 08:40:33 +0300
Subject: [PATCH] [SW-10909] goya2: ppboot: skip pcie init

This patch removes the "Skip PCIe initialization"
capability as no bit is present in Goya2 for skipping
the PCIe initialization either in DumpSel or in BOOT
STRAP register. While these bits get identified,
the PCIe Initialization will always be skipped, to
meet initial power on requirements.

Change-Id: I20d5d5c87b612124eb7f7b3ec07d27e2536f45cc
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/goya2/asic.c b/goya2/asic.c
index 5c904e8..f5d905a 100644
--- a/goya2/asic.c
+++ b/goya2/asic.c
@@ -15,8 +15,11 @@
 
 u32 hl_skip_pcie_init(void)
 {
-	/* Use DumpSel[11] to skip PCIe initialization */
-	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L) & BIT(19);
+	/* No bootstrap bit for goya 2 to skip PCIe initialization-
+	 * TBD: identify boot strap bits for both M2 and Addin card variants.
+	 * Until then, skip PCIe Initialization, to facilitate power on.
+	 */
+	return 1;
 }
 
 u64 hl_pll_get_cpu_cycles_per_usec(void)
